Snap 2236

suranjan opened the pull request:
## Changes proposed in this pull request

If a new server is started, it should fail if  at least one previous persistent server is not up and initialized.
  To do that, at the time of GII of the GFXD_PdxTypes we check from the locator
  if there  were any copy of this known to the locator. At least one server which
  was known to the locator must be up and initialized before any new
  server is allowed to do GII.

  This is required otherwise due to DDLReplay proceeding on a new server and creating empty tables and making other copies of the table invalid if they come up later.

## Patch testing

precheckin 
## ReleaseNotes changes

(Does this change require an entry in ReleaseNotes? If yes, has it been added to it?)

## Other PRs 

(Does this change require changes in other projects- snappydata, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)

Commit:
Test

Commit:
Wait for atleast one persistent server to start

  If a new server is started, it should fail if  at least
  one previous persistent server is not up and initialized.

  This is due to DDLReplay proceeding and creating empty tables and
  making other copies of the table invalid if they come up later.

  To do that, at the time of GII of the GFXD_PdxTypes we check from the locator
  if there  were any copy of this known to the locator. At least one server which
  was known to the locator must be up and initialized before any new
  server is allowed to do GII.

Commit:
Cleaned up the debug/log codes

Commit:
Removed redundant code

Commit:
Merge branch 'snappy/master' into SNAP-2236

Commit:
Added the File left in earlier checkin

Commit:
Merge branch 'snappy/master' into SNAP-2236

Commit:
Added some test changes

Commit:
Merge branch 'SNAP-2236' of https://github.com/SnappyDataInc/snappy-store into SNAP-2236

kneeraj commented on gemfire-core/src/main/java/com/gemstone/gemfire/internal/DataSerializableFixedID.java:
```diff
@@ -805,6 +805,8 @@ public int getDSFID() {
   public static final short UNBLOCK_PERSISTENT_ID_RESPONSE = 2165;
   public static final short WAITING_PERSISTENT_IDS_REQUEST = 2166;
   public static final short WAITING_PERSISTENT_IDS_RESPONSE = 2167;
+  public static final short STARTUP_SEQUENCE_QUERY_REQUEST = 2168;
```
Backward compatibility is taken care of?

kneeraj commented on gemfire-core/src/main/java/com/gemstone/gemfire/distributed/internal/membership/jgroup/JGroupMembershipManager.java:
```diff
@@ -1746,7 +1746,7 @@ public JGroupMembershipManager initialize(
                                                myAddr.preferredForCoordinator(),
                                                MemberAttributes.DEFAULT);
-    
+
```
No change. Just a space.

kneeraj commented on gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java:
```diff
@@ -5659,6 +5668,15 @@ public boolean isUnInitializedMember(InternalDistributedMember member) {
     }
   }
 
+  public boolean isSnappyDataStore(InternalDistributedMember member) {
```
May be just "isDataStore" is fine. I was looking for something specific to snappydata mode and not rowstore mode.

kneeraj commented on gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/PartitionedRegionHelper.java:
```diff
@@ -260,7 +260,7 @@ public static LocalRegion getPRRoot(final Cache cache, boolean createIfAbsent)
       factory.setDataPolicy(DataPolicy.REPLICATE);
       factory.addCacheListener(new FixedPartitionAttributesListener());
-      if (Boolean.getBoolean("gemfire.PRDebug")) {
+      if (true || Boolean.getBoolean("gemfire.PRDebug")) {
```
Temporary? Cleanup?

kneeraj commented on gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/partitioned/RedundancyLogger.java:
```diff
@@ -421,6 +421,8 @@ else if(thereAreBucketsToBeRecovered && !allMembersToWaitFor.isEmpty()){
       else if(!this.loggedDoneMessage) {
         logDoneMessage();
       }
+
```
Unnecessary space.

kneeraj commented on gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/persistence/PersistenceAdvisorImpl.java:
```diff
@@ -81,8 +85,11 @@
+  private boolean DISALLOW_CLUSTER_RESTART_CHECK = Boolean.getBoolean(
+      "gemfire.DISALLOW_CLUSTER_RESTART_CHECK");
+
+  public static final boolean TRACE = true;//Boolean.getBoolean("gemfire.TRACE_PERSISTENCE_ADVISOR");
```
Temporary? hard coded. Remove it.

Commit:
- Fixed the issue of locator properly skipping ddls other than related
  to hive metastore.
- Clearing the System Property FORCE_LOCATOR_DM_TYPE in clearStatics as the
  same vm can be used for a non locator type node in tests later.
- Properly clearing the system property off-heap-size etc. This improper
  cleanup was letting to failure of some off heap tests.
- Cleaning datadictionary files created in LDapGroupAuth test similar to other
  tests as ddls created in this tests were getting replayed in some other tests
  which started in the same jvm.

Commit:
Added list-missing-disk-store back in the list of valid commands.
Addresses SNAP-2300
Now hive metstore related tables can be on locator so removed the
assertion that GfxdTxEntryState cannot be created on locator in Snappy mode only.

Commit:
Merge branch 'snappy/master' into SNAP-2236

Commit:
Merge branch 'snappy/master' into SNAP-2236

Commit:
Merge branch 'snappy/master' into SNAP-2236

Commit:
Did some clean up of logs

sumwale commented:
@suranjan Wouldn't it be better to check for previous owners of DDL region rather than GFXD_PdxTypes? Latter is redundant in Snappy and should ideally be removed.

suranjan commented:
@sumwale Wanted to, but this was first region to be created. Will have to remove this region first, otherwise that causes another set of dependecies because on some servers GFXD_PdxTypes gets initialized with new persistent view.

Cleanup of this region can be done with change of this code to use DDL Region.
